ros2topic/ros2topic/verb/hz.py

```diff
@@ -225,43 +225,98 @@ def get_hz(self, topic=None):
+            rate, min_delta, max_delta, std_dev, window = get_format_hz(ret)
+            print('average rate: %.3f\n\tmin: %.3fs max: %.3fs std dev: %.5fs window: %s'
+                  % (rate, min_delta, max_delta, std_dev, window))
+            return
 
-def _rostopic_hz(node, topic, window_size=DEFAULT_WINDOW_SIZE, filter_expr=None, use_wtime=False):
+        # monitoring multiple topics' hz
+        header = ['topic', 'rate', 'min_delta', 'max_delta', 'std_dev', 'window']
+        stats = {h: [] for h in header}
+        for topic in topics:
+            hz_stat = self.get_hz(topic)
+            if hz_stat is None:
+                continue
+            rate, min_delta, max_delta, std_dev, window = get_format_hz(hz_stat)
+            stats['topic'].append(topic)
+            stats['rate'].append('{:.3f}'.format(rate))
+            stats['min_delta'].append('{:.3f}'.format(min_delta))
+            stats['max_delta'].append('{:.3f}'.format(max_delta))
+            stats['std_dev'].append('{:.5f}'.format(std_dev))
+            stats['window'].append(str(window))
+        if not stats['topic']:
+            return
+        print(_get_ascii_table(header, stats))
+
+
+def _get_ascii_table(header, cols):
+    # compose table with left alignment
+    header_aligned = []
+    col_widths = []
+    for h in header:
+        col_width = max(len(h), max(len(el) for el in cols[h]))
+        col_widths.append(col_width)
+        header_aligned.append(h.center(col_width))
+        for i, el in enumerate(cols[h]):
+            cols[h][i] = str(cols[h][i]).ljust(col_width)
+    # sum of col and each 3 spaces width
+    table_width = sum(col_widths) + 3 * (len(header) - 1)
+    n_rows = len(cols[header[0]])
+    body = '\n'.join('   '.join(cols[h][i] for h in header) for i in range(n_rows))
+    table = '{header}\n{hline}\n{body}\n'.format(
+        header='   '.join(header_aligned), hline='=' * table_width, body=body)
+    return table
+
+
+def _rostopic_hz(node, topics, window_size=DEFAULT_WINDOW_SIZE, filter_expr=None, use_wtime=False):
     """
     Periodically print the publishing rate of a topic to console until shutdown.
 
-    :param topic: topic name, ``list`` of ``str``
+    :param topics: list of topic names, ``list`` of ``str``
     :param window_size: number of messages to average over, -1 for infinite, ``int``
     :param filter_expr: Python filter expression that is called with m, the message instance
     """
     # pause hz until topic is published
-    msg_class = get_msg_class(
-        node, topic, blocking=True, include_hidden_topics=True)
+    rt = ROSTopicHz(node, window_size, filter_expr=filter_expr, use_wtime=use_wtime)
+    topics_len = len(topics)
+    for topic in topics:
+        msg_class = get_msg_class(
+            node, topic, blocking=True, include_hidden_topics=True)
 
-    if msg_class is None:
-        node.destroy_node()
-        return
+        if msg_class is None:
+            node.destroy_node()
+            return
```

Comment:
I believe that what we can do better here is to skip the unknown message classes with printing the warning message for user, and proceed to measure the frequency with found ones.

Reply:
fixed with https://github.com/ros2/ros2cli/pull/929/commits/0d066fd18b55d38a27d000f4404cf86458e99258